--- spm_fmri_spm_ui.m	2004-06-29 11:56:41.000000000 +0200
+++ spm_fmri_spm_ui.m	2009-03-20 00:00:00.000000000 +0100
@@ -380,10 +380,10 @@
 g     = zeros(q,1);
 fprintf('%-40s: %30s','Calculating globals',' ')                     %-#
 for i = 1:q
-	fprintf('%s%30s',sprintf('\b')*ones(1,30),sprintf('%4d/%-4d',i,q)) %-#
+	fprintf('%s%30s',char(sprintf('\b')*ones(1,30)),sprintf('%4d/%-4d',i,q)) %-#
 	g(i) = spm_global(VY(i));
 end
-fprintf('%s%30s\n',sprintf('\b')*ones(1,30),'...done')               %-#
+fprintf('%s%30s\n',char(sprintf('\b')*ones(1,30)),'...done')               %-#
 
 % scale if specified (otherwise session specific grand mean scaling)
 %-----------------------------------------------------------------------
--- spm_spm.m	2005-01-28 11:34:27.000000000 +0100
+++ spm_spm.m	2009-03-20 00:00:00.000000000 +0100
@@ -471,7 +471,7 @@
 nbch     = ceil(xdim*ydim/blksz);				%-# blocks
 
 
-fprintf('%s%30s\n',sprintf('\b')*ones(1,30),'...done')               %-#
+fprintf('%s%30s\n',char(sprintf('\b')*ones(1,30)),'...done')               %-#
 
 
 %-Initialise output images (unless this is a 1st pass for ReML)
@@ -532,7 +532,7 @@
 VResI = spm_create_vol(VResI,'noopen');
 end % (xX,'W')
 
-fprintf('%s%30s\n',sprintf('\b')*ones(1,30),'...initialised')        %-#
+fprintf('%s%30s\n',char(sprintf('\b')*ones(1,30)),'...initialised')        %-#
 
 
 %=======================================================================
@@ -585,7 +585,7 @@
 
 	%-Get data & construct analysis mask
 	%===============================================================
-	fprintf('%s%30s',sprintf('\b')*ones(1,30),'...read & mask data')%-#
+	fprintf('%s%30s',char(sprintf('\b')*ones(1,30)),'...read & mask data')%-#
 	Cm    = logical(ones(1,nVox));			%-current mask
 
 
@@ -633,13 +633,13 @@
 
 		%-Whiten/Weight data and remove filter confounds
 		%-------------------------------------------------------
-		fprintf('%s%30s',sprintf('\b')*ones(1,30),'filtering')	%-#
+		fprintf('%s%30s',char(sprintf('\b')*ones(1,30)),'filtering')	%-#
 
 		KWY   = spm_filter(xX.K,W*Y);
 
 		%-General linear model: Weighted least squares estimation
 		%------------------------------------------------------
-		fprintf('%s%30s',sprintf('\b')*ones(1,30),' estimation') %-#
+		fprintf('%s%30s',char(sprintf('\b')*ones(1,30)),' estimation') %-#
 
 		beta  = xX.pKX*KWY;			%-Parameter estimates
 		res   = spm_sp('r',xX.xKXs,KWY);	%-Residuals
@@ -699,7 +699,7 @@
     %===================================================================
     if isfield(xX,'W')
  
-    	fprintf('%s%30s',sprintf('\b')*ones(1,30),'...saving plane')	%-#
+    	fprintf('%s%30s',char(sprintf('\b')*ones(1,30)),'...saving plane')	%-#
 
 	%-Write Mask image
 	%-------------------------------------------------------------------
@@ -731,7 +731,7 @@
 
     %-Report progress
     %-------------------------------------------------------------------
-    fprintf('%s%30s',sprintf('\b')*ones(1,30),'...done')             %-#
+    fprintf('%s%30s',char(sprintf('\b')*ones(1,30)),'...done')             %-#
     spm_progress_bar('Set',100*(bch + nbch*(z - 1))/(nbch*zdim));
 
 
@@ -748,7 +748,7 @@
 %=======================================================================
 if isfield(xX,'W')
 
-	fprintf('%s%30s\n',sprintf('\b')*ones(1,30),'...closing files')  %-#
+	fprintf('%s%30s\n',char(sprintf('\b')*ones(1,30)),'...closing files')  %-#
 	VM              = spm_close_vol(VM);
 	Vbeta           = spm_close_vol(Vbeta);
 	VResI           = spm_close_vol(VResI);
@@ -895,7 +895,7 @@
 xX.nKX        = spm_DesMtx('sca',xX.xKXs.X,xX.name);
 
 
-fprintf('%s%30s\n',sprintf('\b')*ones(1,30),'...done')               %-#
+fprintf('%s%30s\n',char(sprintf('\b')*ones(1,30)),'...done')               %-#
 
 %-Save remaining results files and analysis parameters
 %=======================================================================
@@ -936,7 +936,7 @@
 %=======================================================================
 %- E N D: Cleanup GUI
 %=======================================================================
-fprintf('%s%30s\n',sprintf('\b')*ones(1,30),'...done')               %-#
+fprintf('%s%30s\n',char(sprintf('\b')*ones(1,30)),'...done')               %-#
 spm('FigName','Stats: done',Finter); spm('Pointer','Arrow')
 fprintf('%-40s: %30s\n','Completed',spm('time'))                     %-#
 fprintf('...use the results section for assessment\n\n')             %-#
